internal/resourcestatus/resourcekey.go

```diff
@@ -63,124 +37,96 @@ func resourceCountsPerCluster(list *fleet.BundleDeploymentList) map[string]*flee
 	return res
 }
 
+type resourceStateEntry struct {
+	fleet.ResourcePerClusterState
+	incomplete bool
+}
+
+type resourceStatesByResourceKey map[fleet.ResourceKey][]resourceStateEntry
+
 // fromResources inspects all bundledeployments for this GitRepo and returns a list of
 // Resources and error messages.
 //
 // It populates gitrepo status resources from bundleDeployments. BundleDeployment.Status.Resources is the list of deployed resources.
```

Comment:
This is not entirely relevant anymore: it should probably mention that the first return value lists resources by cluster.